nursery/persist-via-print-monitors-registry-key.yml

```diff
@@ -0,0 +1,19 @@
+rule:
+  meta:
+    name: persist via Print Monitors registry key
+    namespace: persistence/registry
+    authors:
+      - j.j.vannielen@utwente.nl
+    scopes:
+      static: function
+      dynamic: call
+    att&ck:
+      - Persistence::Boot or Logon Autostart Execution::Port Monitors [T1547.010]
```

Comment:
Should the rule name use _Port Monitors_ instead of _Print Monitors_?

Reply:
Thanks for your input, @mike-hunhoff !

To my understanding, the cape report always contains the registry path and value name within one registry write function call.

E.g.
```
         {
            ...
            "category": "registry",
            "api": "RegSetValueExW",
            "status": true,
            "return": "0x00000000",
            "arguments": [
              {
                "name": "Handle",
                "value": "0x00000228"
              },
              {
                "name": "ValueName",
                "value": "DefaultNotificationsSetting"
              },
              {
                "name": "Type",
                "value": "4",
                "pretty_value": "REG_DWORD"
              },
              {
                "name": "Buffer",
                "value": "1"
              },
              {
                "name": "BufferLength",
                "value": "4"
              },
              {
                "name": "FullName",
                "value": "HKEY_LOCAL_MACHINE\\SOFTWARE\\Policies\\Google\\Chrome\\DefaultNotificationsSetting"
              }
            ],
            ...
          },
```

In what case do the registry path and value name spread out over multiple Windows API calls?

Reply:
I'm not familiar with the underlying representation used by CAPE for API calls, but it appears that it may be adding some syntactic sugar e.g. _FullName_ is a field specific to CAPE.

Other sandboxes may not do this, e.g. VMRay:
```xml
<fncall ts="126662" fncall_id="28904" process_id="1" thread_id="1" name="RegOpenKeyExW" addr="0x7ff8e7439670" from="0x7ff8a1570edf">
	<in>
		<param name="hKey" type="void_ptr" value="0xffffffff80000001"/>
		<param name="lpSubKey" type="ptr" value="0x2c7d58c">
			<deref type="str" value="Software\\Microsoft\\Windows\\CurrentVersion\\Run"/>
		</param>
		<param name="ulOptions" type="unsigned_32bit" value="0x0"/>
		<param name="samDesired" type="unknown" value="0x2001f"/>
		<param name="phkResult" type="void_ptr" value="0xbbdb78"/>
	</in>
	<out>
		<param name="phkResult" type="ptr" value="0xbbdb78">
			<deref type="unsigned_64bit" value="0x75c"/>
		</param>
		<param name="ret_val" type="unknown" value="0x0"/>
	</out>
</fncall>
[...]
<fncall ts="126664" fncall_id="28906" process_id="1" thread_id="1" name="RegSetValueExW" addr="0x7ff8e743b080" from="0x7ff8a1574d2c">
	<in>
		<param name="hKey" type="void_ptr" value="0x75c"/>
		<param name="lpValueName" type="ptr" value="0x2c7b5b4">
			<deref type="str" value="Windows Defender Updater"/>
		</param>
		<param name="Reserved" type="unsigned_32bit" value="0x0"/>
		<param name="dwType" type="unsigned_32bit" value="0x1"/>
		<param name="lpData" type="ptr" value="0x2c7d33c">
[...]
		</param>
		<param name="cbData" type="unsigned_32bit" value="0x7a"/>
	</in>
[...]
</fncall>
```

You can see the subkey `Software\Microsoft\Windows\CurrentVersion\Run` and value `Windows Defender Updater` are referenced in separate API calls `RegOpenKeyExW` and `RegSetValueExW`, respectively (`0x75c` is the handle to the opened registry subkey).

Although this does not appear to be the case for CAPE, we assume most sandboxes will log API calls, including parameter and return names and values, according to their documentation, e.g. https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regsetvalueexw.

@mr-tz do you know if the CAPE extractor is extracting the _FullName_ field?

Reply:
I think CAPE is doing some enhancements to the API trace by enhancing values. We cannot assume this for all sandbox backends so to be flexible rules should account for that.

Reply:
Good discussion here!

> For much better handling moving forward we need a different approach (like a new scope within a certain range, https://github.com/mandiant/capa-rules/discussions/951#discussioncomment-11112001).

Agree.

> I expect that for some rules, changing the scope from call to threat will result in many false positives. I think that as a result we will have to remove these rules, which is a pitty.

In spirit, I agree with you. Though, I'm curious about this in practice. It *seems* like there might be FPs, but could we test this a bit? Maybe the behavior isn't quite as bad as we worry about.

I only suggest this (which doesn't *feel* very good) because we think we have a plan for a better implementation (#951) so many a little inaccuracy in the meantime is acceptable.

Reply:
> That raises another issue with capa's existing rule set registry value. The dynamic scope for this rule should also be thread instead of call otherwise the [optional match for create or open registry key](https://github.com/mandiant/capa-rules/blob/ed816a8e53446cd21a9d634b7d0531df664ba1c1/host-interaction/registry/create/set-registry-value.yml#L22) can never match (it's optional so it's not show stopper). Thoughts @mr-tz ?

@mike-hunhoff https://github.com/mandiant/capa/issues/2124 raises the issue that capa doesn't validate scope dependencies correctly, so there are probably more of these hidden mistakes lurking around.

In the meantime, I agree this isn't a showstopper, since it's only an optional term that can't match. We have a bit of work to do to get all these scopes straightened out...